src/movies.js

```diff
@@ -1,16 +1,69 @@
+
+
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
+
+
+function getAllDirectors(moviesArr) {
+  const totalDirectors = movies.map(function(filmObj){
+    return filmObj.director;
+  });
+  console.log(totalDirectors);
+}
+
+
+
+
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies() {}
+function howManyMovies(arr) {
+  const result = movies.filter(function (film) {
+    return film.director === 'Steven Spielberg' && film.genre.includes('Drama');
+      
+  });
+  console.log(result.length);
+}
+  
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage() {}
+function scoresAverage(arr) {
+  if(!movies.length){
+    return 0;
+  }
+  const result = movies.map(function (film) {
+    return film.score;
+  });
+  
+  const sum = movies.reduce(function(acc, currentValue){
+    return acc + currentValue;
+  }, 0); 
```

Comment:
When working with objects, pay attention if you need to access full object or its property. In this case you only want to be adding score. 

Movies -> it is array of movie objects
CurrentValue -> is single movie object

So what you want to be adding  is: acc + currentValue.score